src/formatDate.js

```diff
@@ -8,7 +8,50 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  let year;
+  let month;
+  let day;
+  const formattedDate = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YYYY') {
+      year = dateParts[i];
+    }
+
+    if (fromFormat[i] === 'YY') {
+      year = dateParts[i] < 30 ? `20${dateParts[i]}` : `19${dateParts[i]}`;
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = dateParts[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      day = dateParts[i];
+    }
+  }
+
+  for (const format of toFormat) {
+    switch (format) {
+      case 'YYYY':
+        formattedDate.push(year);
+        break;
+
+      case 'YY':
+        formattedDate.push(year.slice(-2));
+        break;
+
+      case 'MM':
+        formattedDate.push(month);
+        break;
+
+      case 'DD':
+        formattedDate.push(day);
+    }
+  }
+
+  return formattedDate.join(toFormat[3]);
```

Comment:
The implementation of the `formatDate` function seems correct. However, there are a couple of improvements that could be made for better readability and performance:
- Instead of using a `for` loop to iterate over `fromFormat`, you could use object destructuring to assign `year`, `month`, and `day` directly if the order of the date parts is known to be consistent with the `fromFormat` array.
- The second loop that builds `formattedDate` can be simplified by mapping over `toFormat` and applying the transformation directly within the `map` function, which would eliminate the need for the `switch` statement and make the code more concise.